src/function/scalar/compressed_materialization/compress_integral.cpp

```diff
@@ -0,0 +1,226 @@
+	D_ASSERT(args.data[1].GetType() == result.GetType());
+	const auto min_val = ConstantVector::GetData<RESULT_TYPE>(args.data[1])[0];
+	UnaryExecutor::Execute<INPUT_TYPE, RESULT_TYPE>(args.data[0], result, args.size(), [&](const INPUT_TYPE &input) {
+		return TemplatedIntegralDecompress<INPUT_TYPE, RESULT_TYPE>(input, min_val);
+	});
+}
+
+template <class INPUT_TYPE, class RESULT_TYPE>
+static scalar_function_t GetIntegralDecompressFunction(const LogicalType &input_type, const LogicalType &result_type) {
+	return IntegralDecompressFunction<INPUT_TYPE, RESULT_TYPE>;
+}
+
+template <class INPUT_TYPE>
+static scalar_function_t GetIntegralDecompressFunctionResultSwitch(const LogicalType &input_type,
+                                                                   const LogicalType &result_type) {
+	switch (result_type.id()) {
+	case LogicalTypeId::SMALLINT:
+		return GetIntegralDecompressFunction<INPUT_TYPE, int16_t>(input_type, result_type);
+	case LogicalTypeId::INTEGER:
+		return GetIntegralDecompressFunction<INPUT_TYPE, int32_t>(input_type, result_type);
+	case LogicalTypeId::BIGINT:
+		return GetIntegralDecompressFunction<INPUT_TYPE, int64_t>(input_type, result_type);
+	case LogicalTypeId::HUGEINT:
+		return GetIntegralDecompressFunction<INPUT_TYPE, hugeint_t>(input_type, result_type);
+	case LogicalTypeId::USMALLINT:
+		return GetIntegralDecompressFunction<INPUT_TYPE, uint16_t>(input_type, result_type);
+	case LogicalTypeId::UINTEGER:
+		return GetIntegralDecompressFunction<INPUT_TYPE, uint32_t>(input_type, result_type);
+	case LogicalTypeId::UBIGINT:
+		return GetIntegralDecompressFunction<INPUT_TYPE, uint64_t>(input_type, result_type);
+	default:
+		throw InternalException("Unexpected input type in GetIntegralDecompressFunctionSetSwitch");
+	}
+}
+
+static scalar_function_t GetIntegralDecompressFunctionInputSwitch(const LogicalType &input_type,
+                                                                  const LogicalType &result_type) {
+	switch (input_type.id()) {
+	case LogicalTypeId::UTINYINT:
+		return GetIntegralDecompressFunctionResultSwitch<uint8_t>(input_type, result_type);
+	case LogicalTypeId::USMALLINT:
+		return GetIntegralDecompressFunctionResultSwitch<uint16_t>(input_type, result_type);
+	case LogicalTypeId::UINTEGER:
+		return GetIntegralDecompressFunctionResultSwitch<uint32_t>(input_type, result_type);
+	case LogicalTypeId::UBIGINT:
+		return GetIntegralDecompressFunctionResultSwitch<uint64_t>(input_type, result_type);
+	default:
+		throw InternalException("Unexpected result type in GetIntegralDecompressFunctionInputSwitch");
+	}
+}
+
+template <CompressedMaterializationDirection DIRECTION>
+static void CMIntegralSerialize(FieldWriter &writer, const FunctionData *bind_data_p, const ScalarFunction &function) {
+	writer.WriteField(DIRECTION);
+	writer.WriteSerializable(function.return_type);
+	writer.WriteRegularSerializableList(function.arguments);
```

Comment:
The child expressions and return type should already be (de)serialized as part of the generic function deserialization - and shouldn't `DIRECTION` be encoded in the function name?

As a general rule most functions should not need to implement the (de)serialize functions unless they do "special" things (like e.g. the regex function that contains a pre-compiled regex in the function data).

Reply:
We need to do the (de)serialization because we cannot get the function using the function name, as this would require a bind. The compressed materialization functions cannot be bound because they are for internal use only, and we throw an error there.

I've used `DIRECTION` for convenience, but we can definitely do without it. I'll clean this up.